File: practices/practice4.1/src/PizzaDirector.cpp
#include "../include/PizzaDirector.hpp"

PizzaDirector::PizzaDirector(std::unique_ptr<PizzaBuilder>& builder) : pizzaBuilder(builder) {}

std::unique_ptr<Pizza> PizzaDirector::makePizza() {
    return pizzaBuilder->getPizza();
}

File: homeworks/homework4.1/src/main/java/com/pizza/PizzaDirector.java
package com.pizza;

public class PizzaDirector {
    private final PizzaBuilder pizzaBuilder;

    public PizzaDirector(PizzaBuilder pizzaBuilder) {
        this.pizzaBuilder = pizzaBuilder;
    }

    public Pizza makePizza() {
        return pizzaBuilder.getPizza();
    }
}
